src/graph/bipartite.h

```diff
@@ -171,6 +178,517 @@ class Bipartite : public BaseHeteroGraph {
   COOPtr coo_;
 };
 
+//////////////////////////////////////////////////////////
+//
+// COO graph implementation
+//
+//////////////////////////////////////////////////////////
```

Comment:
Why move them in header file?

Reply:
Because I need to include them in `src/kernel/binary_reduce.h`.